frontend/main.go

```diff
@@ -25,8 +25,19 @@ type Config struct {
 	AOLEnabled       bool   `json:"aolenabled"`
 }
 
+// Used to request databases from the global list.
+type DBOpenRequest struct {
+	// Name of the database requested:
+	DBName string
+	// Filled out on return:
+	Database goleg.Database
+	DBError error
+	SenderChannel chan DBOpenRequest
+}
```

Comment:
This feels like an abusive way to do things. Maybe you should consider an Event -> Action pair instead of using one struct?

```go
// We can infer that this is for a databae by the package name.
type OpenEvent struct {
    DBName string // Name of the database requested
    Reply chan OpenAction // Reply channel
}

type OpenAction struct {
    Database goleg.Database
    Err error
}
```

Much cleaner, no?